app/ai/TweetGeneratorOpenAI.py

```diff
@@ -1,21 +1,45 @@
+import logging
+
 from openai import OpenAI
 
-from app.ai.models import TweetModel, TweetThreadModel
+from app.ai.agents.CryptoMarketAnalysisFormatAgent import CryptoMarketAnalysisFormatAgent
+from app.ai.models import (
+    TweetModel,
+    TweetThreadModel,
+    CryptoAnalysisThreadModel
+)
 from config.prompts import (
     SYSTEM_PROMPT,
     USER_PROMPT_TWITTER,
     TWITTER_PROMPT_SINGLE_TWEET,
     TWITTER_PROMPT_THREAD,
     TWITTER_PROMPT_REPLY,
+    CRYPTO_SYSTEM_PROMPT,
+    get_analysis_prompt
 )
 from app.utils.utils import format_tweet_timeline
+from app.core.exceptions import (
+    TweetGenerationError,
+    MarketDataError,
+)
+from app.ai.agents.ToneAgent import ToneAgent
+
+logger = logging.getLogger(__name__)
 
 
 class TweetGeneratorOpenAI:
     def __init__(self, api_key: str):
+        """
+        Initialize the tweet generator with OpenAI API key
+        
+        Args:
+            api_key (str): OpenAI API key for authentication
+        """
         self.system = SYSTEM_PROMPT
+        self.crypto_system = CRYPTO_SYSTEM_PROMPT
         self.prompt = USER_PROMPT_TWITTER
         self.client = OpenAI(api_key=api_key)
+        self.model = "gpt-4"
```

Comment:
This is set but never used. The correct model is already hardcoded as `gpt-4o-mini`.